### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by QuentinBisson**
Towards https://github.com/giantswarm/giantswarm/issues/25396

This rewrites the documentation for Promtail.

This:
- Adds an upgrade section in the Readme to explain how to migrate from 0.x to 1.x
- Move TESTING.md to a CONTRIBUTING.md file and adding an how to upgrade section for us.
- Upgrades the values.schema.json
- Fix the sample config.

I will not do in the the first phase but I think adding configuration samples based on https://github.com/grafana/helm-charts/blob/main/charts/promtail/README.md#configuration or on customer needs

### CODE REVIEW

File: sample_configs/values-gs.yaml [hunk 1]
```diff
@@ -1,86 +1,59 @@
...
+  # make promtail print config on start
+  extraArgs:
+    - -log-config-reverse-order
+
+  # Information about the Loki endpoint to connect to. Currently the chart supports
+  # only a single upstream connection.
+  config:
+    clients:
+    - url: "https://loki.m2h60.k8s.gauss.eu-west-1.aws.gigantic.io/loki/api/v1/push"
+      tenant_id: 1
```
Comment (hervenicol) [inline sample_configs/values-gs.yaml:12]: We could set something more meaningful there.
```suggestion
      tenant_id: gorilla-loki0
```

**Review by hervenicol — APPROVED** [review]
Great job! :+1:

**Comment by QuentinBisson** [comment]
Nice :) ideally i'd like a session on multi-tenancy that would be cross referenced in the loki section as well but I don't have enough data on that yet

**Comment by QuentinBisson** [comment]
@TheoBrigitte you want to check the current state? Then I can merge and improve with examples tomorrow

**Comment by QuentinBisson** [comment]
I'll try to follow the same structure on all repos

**Review by QuantumEnigmaa — APPROVED** [review]

**Review by LolloneS — APPROVED** [review]

**Pull request merged**